### ISSUE TITLE
stroke-dasharray not supported on path elements

### ISSUE THREAD

**Issue by stephenanders** [canvg/canvg#454]
My stroke-dasharray is not being rendered by canvg b/c it is on a path element rather than a line element.  My SVG is being generated using highcharts, so I dont really have the option of switching to a line.

According to [Mozilla](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-dasharray) stroke-dasharray should be supported in SVG for the following element types:
**Shape Elements**
`<circle>, <ellipse>, <line>, <path>, <polygon>, <polyline>, <rect>`
**Text Content Elements**
`<altGlyph>, <altGlyphDef>, <altGlyphItem>, <glyph>, <glyphRef>, <textPath>, <text>, <tref>, <tspan>`

Demo SVG showing the issue:

``` svg
<svg xmlns="http://www.w3.org/2000/svg" width="200" height="200" viewPort="0 0 200 300" version="1.1">
    <line stroke-dasharray="5, 2" x1="10" y1="10" x2="100" y2="10" style="stroke:darkgreen;stroke-width:4" />
    <path fill="none" stroke-dasharray="5,2" d="M 10 20 L 100 20 z" style="stroke:darkgreen;stroke-width:4" />
</svg>
```

**Comment by stephenanders** [comment]
created a pull request with the fix for this issue #455

**Comment by alafr** [comment]
The `closePath()` call is not guilty (the "Z" command should be removed from the path attribute instead?). But there is a bug with dashes: canvg uses an default dash-offset of 1px instead of 0. Another example shows that the problem does not affect only paths:
```
<svg xmlns="http://www.w3.org/2000/svg">
  <g transform="scale(6) translate(4,6)">
    <path stroke="red" stroke-width="4" stroke-dasharray="4,4" d="M0,0 L32,0"/>
    <path stroke="blue" stroke-width="4" stroke-dasharray="4,4" d="M32,4 L0,4"/>
    <line stroke="red" stroke-width="4" stroke-dasharray="4,4" x1="0" y1="8" x2="32" y2="8"/>
    <line stroke="blue" stroke-width="4" stroke-dasharray="4,4" x1="32" y1="12" x2="0" y2="12"/>
  </g>
</svg>
```
![image](https://cloud.githubusercontent.com/assets/17966696/19207165/0b388b10-8cf7-11e6-9e13-7e93d46a3fbc.png)

https://github.com/canvg/canvg/blob/master/canvg.js#L979
https://www.w3.org/TR/SVG11/painting.html#StrokeDashoffsetProperty
